### PR TITLE
[WIP] JSON

### PR DESCRIPTION
**Pull request by jbrooksuk**
Working towards #203 

---

Firstly, we have to start the long and tedious process of converting the README.md file to a JSON list. I've completed two categories, but I wanted to get feedback on the format I've used.

I've also used `gulp` and `gulp-jsonlint` because this can become part of the CI process for new contributions.

### CODE REVIEW

**Comment by jbrooksuk** [comment]
@yannbertrand probably 😄 it's more to discuss the structure before we generate it.

**Comment by yannbertrand** [comment]
Yup! We should probably check out how github-pages work. I think we need to put that in a `gh-pages` branch if we decide to use it

**Comment by jbrooksuk** [comment]
You can tell GitHub to use a specific branch or a `docs` directory.

**Comment by yannbertrand** [comment]
Good to know 👍

The format seems fine to me 😉

**Comment by mikestreety** [comment]
I would personally put the category as an attribute on each API (it could then expand to have many cateogories if required). Apart from that, it looks good.

**Comment by jbrooksuk** [comment]
> I would personally put the category as an attribute on each API (it could then expand to have many cateogories if required). Apart from that, it looks good.

Perhaps an array of categories?

**Comment by mikestreety** [comment]
Yeah, that's what I was thinking.

**Comment by yannbertrand** [comment]
```
[
    {
        "name": "IUCN",
        "description": "IUCN Red List of Threatened Species",
        "auth": "token",
        "link": "http://apiv3.iucnredlist.org/api/v3/docs",
        "categories": ["Animals"]
    }, {
        "name": "Petfinder",
        "description": "Adoption",
        "auth": "no",
        "link": "https://www.petfinder.com/developers/api-docs/",
        "categories": ["Animals"]
    }, {
        "name": "RescueGroups",
        "description": "Adoption",
        "auth": "no",
        "link": "https://userguide.rescuegroups.org/display/APIDG/API+Developers+Guide+Home",
        "categories": ["Animals"]
    }, {
        "name": "AniList",
        "description": "AniList Anime API",
        "auth": "oAuth",
        "link": "http://anilist-api.readthedocs.org/en/latest/#",
        "categories": ["Anime"]
    }, {
        "name": "Hummingbird",
        "description": "Hummingbird Anime API",
        "auth": "no",
        "link": "https://hummingbird.me/",
        "categories": ["Anime"]
    }
]
```

**Comment by jbrooksuk** [comment]
Better 👍

**Comment by davemachado** [comment]
While we are laying the structure down for the JSON, should we also add the HTTPS-support as an attribute? Adding that specification is another open issue right now, so it might be good just to try to future-proof this structure as we go.

**Comment by mikestreety** [comment]
I was just thinking about that. I agree.

**Comment by jbrooksuk** [comment]
For sure!

**Comment by davemachado** [comment]
Discussion for this carries on in #368

File: apis.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                "link": "https://www.petfinder.com/developers/api-docs/"
+            }, {
+                "name": "RescueGroups",
+                "description": "Adoption",
+                "auth": "no",
+                "link": "https://userguide.rescuegroups.org/display/APIDG/API+Developers+Guide+Home"
+            }
+        ]
+    }, {
+        "category": "Anime",
```
Comment (kuhnrl30) [inline apis.json]: Wanted to point out that the JSON format used here is different than what was last recommended in #259: 
```
[
    {
        "name": "IUCN",
        "description": "IUCN Red List of Threatened Species",
        "auth": "token",
        "link": "http://apiv3.iucnredlist.org/api/v3/docs",
        "categories": ["Animals"]
    },
```
The difference being how the category is used.  The first format opens us to duplicate entries if an API is put in multiple categories creating a risk that they aren't identical.  i.e. the link gets updated in one entry but not the duplicate.  

Are we ok with that?  I just recently started following so I'm not sure what the protocol is for making decisions on this project.

**Comment by davemachado** [comment]
As far I can tell, we work to limit entries to one category that best fits
the service offered. My opinion is to keep it that way so we don't end up
with a huge list of duplicate entries (for example, Instragram could end up
in Social, Photography, Development, etc since one could argue their API
offers some type of service for each of those categories. We can change the
structure of how entries are entered into this collection, but I fear the
list will become too cluttered if we allow duplicate entries.

**Pull request closed without merging**